### PR TITLE
NEXT-319: Rebasing from `master` to avoid conflicts

### PR DESCRIPTION
**Pull request by SamRussellMM**
- Summary v2 endpoint into Messaging Reports group
- Insight v2 endpoint into Messaging Reports group
- Schemas created for response objects.

### CODE REVIEW

**Review by alexdia25 — APPROVED** [review]

File: spec/openapi.yaml [hunk 1]
```diff
@@ -822,6 +822,306 @@ paths:
...
+          in: query
+          description: Filter results by destination address.
+          style: form
+          explode: true
+          schema:
+            type: string
+            example: +61555555555
+        - name: metadata_key
+          in: query
+          description: Filter results for messages that include a metadata key. If this parameter is included in the query, then either the metadata_value or metadata_values parameter must also be included to filter the report by the appropriate key for the specified value.
```
Comment (liam-herbert-messagemedia) [inline spec/openapi.yaml]: No `metadata_values` in summary reports but there is a second option to enable the `metadata_key` filter which is to _use `metadata_value` as the `group_by` option_

File: spec/openapi.yaml [hunk 2]
```diff
@@ -822,6 +822,306 @@ paths:
...
```
Comment (liam-herbert-messagemedia) [inline spec/openapi.yaml]: Same as above applies here please Sam 🙂

**Review by liam-herbert-messagemedia — CHANGES_REQUESTED** [review]

File: spec/openapi.yaml (hunk 1 above)
Comment (liam-herbert-messagemedia) [reply spec/openapi.yaml]: Hey Sam, I think this still applies as `metadata_value` can either be applied as a filter (as you have described currently) or alternatively be used as the `group_by` option.

**Review by liam-herbert-messagemedia — APPROVED** [review]
Nice one Sam 👍

**Pull request merged**